drivers/frequency/adf4382/adf4382.c

```diff
@@ -0,0 +1,1303 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include "adf4382.h"
+#include "no_os_alloc.h"
+#include "no_os_delay.h"
+#include "no_os_error.h"
+#include "no_os_print_log.h"
+
+/* Charge pump current values expressed in uA */
+static const int adf4382_ci_ua[] = {
+	700,
+	900,
+	1100,
+	1300,
+	1400,
+	1800,
+	2200,
+	2500,
+	2900,
+	3600,
+	4300,
+	5000,
+	5800,
+	7200,
+	8600,
+	11100
+};
+
+/**
+ * @brief Writes data to ADF4382 over SPI.
+ * @param dev	   - The device structure.
+ * @param reg_addr - The register address.
+ * @param data 	   - Data value to write.
+ * @return 	   - 0 in case of success or negative error code otherwise.
+ */
+int adf4382_spi_write(struct adf4382_dev *dev, uint16_t reg_addr, uint8_t data)
+{
+	uint8_t buff[ADF4382_BUFF_SIZE_BYTES];
+	uint16_t cmd;
+
+	if (!dev)
+		return -EINVAL;
+
+	cmd = ADF4382_SPI_WRITE_CMD | reg_addr;
+	if (dev->spi_desc->bit_order) {
+		buff[0] = no_os_bit_swap_constant_8(cmd & 0xFF);
+		buff[1] = no_os_bit_swap_constant_8(cmd >> 8);
+		buff[2] = no_os_bit_swap_constant_8(data);
+	} else {
+		buff[0] = cmd >> 8;
+		buff[1] = cmd & 0xFF;
+		buff[2] = data;
+	}
+
+	return no_os_spi_write_and_read(dev->spi_desc, buff,
+					ADF4382_BUFF_SIZE_BYTES);
+}
+
+/**
+ * @brief Reads data from ADF4382 over SPI.
+ * @param dev 	   - The device structure.
+ * @param reg_addr - The register address.
+ * @param data	   - Data read from the device.
+ * @return	   - 0 in case of success or negative error code otherwise.
+ */
+int adf4382_spi_read(struct adf4382_dev *dev, uint16_t reg_addr, uint8_t *data)
+{
+	int ret;
+	uint8_t buff[ADF4382_BUFF_SIZE_BYTES];
+	uint16_t cmd;
+
+	if (!dev)
+		return -EINVAL;
+
+	cmd = ADF4382_SPI_READ_CMD | reg_addr;
+	if (dev->spi_desc->bit_order) {
+		buff[0] = no_os_bit_swap_constant_8(cmd & 0xFF);
+		buff[1] = no_os_bit_swap_constant_8(cmd >> 8);
+		buff[2] = no_os_bit_swap_constant_8(ADF4382_SPI_DUMMY_DATA);
+	} else {
+		buff[0] = cmd >> 8;
+		buff[1] = cmd & 0xFF;
+		buff[2] = ADF4382_SPI_DUMMY_DATA;
+	}
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, buff,
+				       ADF4382_BUFF_SIZE_BYTES);
+	if(ret)
```

Comment:
please follow the linux guidelines -> `if (ret)` Check the entire code since there are multiple occurences. Same applies for other statements (`for`, `while`, `switch` etc).